src/cpp/rtps/builtin/discovery/participant/timedevent/RemoteParticipantLeaseDuration.cpp

```diff
@@ -59,24 +61,38 @@ void RemoteParticipantLeaseDuration::event(EventCode code, const char* msg)
 
     if(code == EVENT_SUCCESS)
     {
-        logInfo(RTPS_LIVELINESS,"RTPSParticipant no longer ALIVE, trying to remove: "
-                << mp_participantProxyData->m_guid);
+        // Check last received message's time_point plus lease duration time doesn't overcome now().
+        // If overcame, remove participant.
+        auto now = std::chrono::steady_clock::now();
+        auto real_lease_tm = last_received_message_tm +
+                std::chrono::microseconds(TimeConv::Duration_t2MicroSecondsInt64(mp_participantProxyData->m_leaseDuration));
```

Comment:
It may be a good idea to make this calculation in the constructor, and have a method to update it (in case the QoS is changeable)

Reply:
Done